src/solvers/flattening/boolbv_byte_update.cpp

```diff
@@ -9,17 +9,18 @@ Author: Daniel Kroening, kroening@kroening.com
 #include "boolbv.h"
 
 #include <iostream>
-#include <cassert>
 
 #include <util/arith_tools.h>
 #include <util/byte_operators.h>
+#include <util/invariant.h>
 
 #include "bv_endianness_map.h"
 
 bvt boolbvt::convert_byte_update(const byte_update_exprt &expr)
 {
-  if(expr.operands().size()!=3)
-    throw "byte_update takes three operands";
+  DATA_INVARIANT(
+    expr.operands().size() == 3,
+    "byte_update expression shall have three operands");
```

Comment:
That's guaranteed by virtue of it being a `byte_update_exprt`, just remove this check.